docs/data-modeling.md

```diff
@@ -0,0 +1,100 @@
+
+#### Try to read a single partition as much as possible
+Because of the data model characteristics, single partition lookup is most efficient. If you need to issue a scan or select request that requires multi-partition lookups or scans ([with cross-partition scan enabled](https://scalardb.scalar-labs.com/docs/latest/configurations/#cross-partition-scan-configurations)), make sure you know what you are doing and think about updating the schemas if possible. For example, in a banking application, if you choose email as a partition key and an account ID as a clustering key and issue a query that specifies an account ID, the query will span all the partitions because it cannot identify the corresponding partition efficiently. In such a case, you should always look up the table with an account ID.
+
+:::note
+
+If you read multiple partitions on a relational database with proper indexes, your query might be efficient because the query is pushed down to the database. 
+
+:::
+
+#### Avoid using secondary indexes as much as possible
+
+Similarly to the above, if you need to issue a scan or select request that uses a secondary index, the request will span all the partitions of a table. Therefore, you should avoid using secondary indexes as much as possible. If you need to use it, use it through a low-selectivity query, which looks up a small portion.
+
+As an alternative to secondary indexes, you can create another table that works as a (clustered) index of a base table. For example, assume there is a table with 3 columns, `table1(A, B, C)`, with a primary key `A`. Then, you can create a table like `index-table1(C, A, B)` with `C` as a primary key so that you can look up a single partition by specifying a value for `C`.
+This approach could speed up read queries but create more load to write queries because it writes to two tables with ScalarDB transactions.
+
+:::note
+
+We plan to have a table-based secondary index feature in the future.
+
+:::
+
+#### Care about data is (assumed to be) distributed by hashing
+
+In the current ScalarDB data model, data is assumed to be distributed by hashing, so you can't perform range queries efficiently without a partition key.
+If you want to issue range queries efficiently, you need to do it within a partition.
+However, it's important to note that this approach requires you to specify a partition key. This can pose scalability issues as the range queries always go to the same partition, potentially overloading it.
+The limitation is not specific to ScalarDB but to databases where data is distributed by hashing for scalability.
```

Comment:
```suggestion
In the current ScalarDB data model, data is assumed to be distributed by hashing. Therefore, you can't perform range queries efficiently without a partition key.

If you want to issue range queries efficiently, you need to do so within a partition. However, if you follow this approach, you must specify a partition key. This can pose scalability issues as the range queries always go to the same partition, potentially overloading it. This limitation is not specific to ScalarDB but to databases where data is distributed by hashing for scalability.
```